### ISSUE TITLE
Multi-architecture support?

### ISSUE THREAD

**Issue by steveeJ** [intermezzOS/kernel#116]
I view this one of the best features in an OS kernel, as it allows reusing anything above the hardware abstracting layer on more than one architecture.
Is multi-architecture support something this project aims for?

**Comment by steveklabnik** [comment]
Not really, or at least, not until far in the future. Given this is mostly about education, learning the details of multiple platforms is a distraction from the primary task, which is an overview of OSes as a whole, IMO.

**Comment by steveeJ** [comment]
My following point is true for the book and the code: intermixing x86(-64) hardware specifics with general OS concepts is more distracting than establishing a clear abstraction.
It is the reality that there is a lot of hardware out there, and there are even more related software specifications that are there to either be adopted or ignored :-D When designing and implementing an OS, the first question that will come up is of course which hardware you target (first). My understanding is that the whole point of an OS, as opposed to a firmware application, is to abstract the hardware bits for most software components so that they can be reused.
